SOAR-6605 Unlock AD/LDAP User from Microsoft Teams (#463)
rapid7/insightconnect-workflows: workflows/Unlock_User_Account_in_Active_Directory_from_Microsoft_Teams/help.md

# Description

> After locking down a user account, it may be convenient to allow administrators to unlock users account. The normal unlock account would require logging into the AD server and manually unlocking it. This automation should save about 3 minutes every time it is used!

Reviewer: ```suggestion
After locking down a user account, it may be convenient to allow administrators to unlock users account. The normal unlock account would require logging into the AD server and manually unlocking it.
```

I lean toward suggesting that we do not provide specific numbers for time saved estimates, but I am not strongly committed to this and can easily be persuaded otherwise to keep it as is or with phrasing tweaks.

Sample Microsoft Teams Trigger Commands:

`!unlock-user-ad <exampleuser>`

# Key Features

* Unlock a user in Active Directory from Microsoft Teams

# Requirements

* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
* Active Directory connection

# Documentation

## Setup

Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.

Once the workflow has been imported, **The `Find DN` step and each Microsoft Teams step will need to be updated to suit your Microsoft Teams and LDAP environments.** Edit the input with the preset text of `change_me` in each of these steps in the workflow.

> 

Reviewer 2: Let's update the workflow to use workflow parameters, and then change this paragraph to tell the user to configure parameters.

# Description

> After locking down a user account, it may be convenient to allow administrators to unlock users account. The normal unlock account would require logging into the AD server and manually unlocking it. 

Reviewer 2: ```suggestion
After locking down a user account, it may be convenient to allow administrators to unlock users accounts. The normal unlock account process would require logging into the AD server and manually unlocking it. This workflow allows a user to unlock an account via Microsoft Teams command.
```

# Description

After locking down a user account, it may be convenient to allow administrators to unlock users account. The normal unlock account would require logging into the AD server and manually unlocking it. 

Sample Microsoft Teams Trigger Commands:

`!unlock-user-ad <exampleuser>`

# Key Features

* Unlock a user in Active Directory from Microsoft Teams

# Requirements

* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
* Active Directory connection

# Documentation

## Setup

Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.

> Once the workflow has been imported, you must edit the workflow paramaters to match you Teams channel and team name, as well as the search base for your Active directory/LDAP instance (see troubleshooting for tips on setting search base). Edit the input with the preset text of `change_me` to the corresponding values for your workflow.

Reviewer 2: ```suggestion
Once the workflow has been imported, you must edit the workflow parameters to match your Teams channel and team name, as well as the search base for your Active Directory/LDAP instance (see troubleshooting for tips on setting search base). 
```

Setting the search base for your instance isn't very intuitive to someone unfamiliar with AD. The search base identifies your organization and folder where the users are kept so that an entire query can be performed. For example, if you had a folder called "Users" in your AD instance, and your AD address was examplecompany.com, your search base would be:
OU=Users,DC=examplecompany,DC=com
If you have multiple "." in the address, split each into its own "DC" section. 

> After configuring the Teams steps, activate the workflow in order to trigger it.

Reviewer 2: ```suggestion
```
